src/builtins/core/date.rs

```diff
@@ -180,6 +180,62 @@ impl PartialDate {
 }
 
 /// The native Rust implementation of `Temporal.PlainDate`.
+///
+/// Represents a calendar date like "March 15th, 2024" without any time or timezone
+/// information. Useful for dates where the specific time of day doesn't matter,
+/// such as deadlines, birth dates, or historical events.
+///
+/// Uses the ISO 8601 calendar (standard Gregorian calendar) by default, with
+/// support for other calendar systems when needed.
+///
+/// ## Examples
+///
+/// ```rust
+/// use temporal_rs::{PlainDate, Calendar};
+///
+/// // Create a date using the ISO calendar
+/// let christmas = PlainDate::try_new_iso(2024, 12, 25).unwrap();
+///
+/// // Explicit calendar specification
+/// let date = PlainDate::try_new(2024, 12, 25, Calendar::default()).unwrap();
```

Comment:
nit: add an assert to test the dates are the same.

Obviously, these are basically the same constructors, but adding the assert would be nice to display that.